sdk/storage/src/common/common_request_policy.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "common/common_request_policy.hpp"
+
+#include <ctime>
+
+namespace Azure { namespace Storage {
+
+  std::unique_ptr<Core::Http::Response> CommonRequestPolicy::Send(
+      Core::Context& ctx,
+      Core::Http::Request& request,
+      Core::Http::NextHttpPolicy nextHttpPolicy) const
+  {
+
+    const auto& headers = request.GetHeaders();
+    if (headers.find("Date") == headers.end() && headers.find("x-ms-date") == headers.end())
+    {
+      // add x-ms-date header in RFC1123 format
+      time_t t = std::time(nullptr);
+      struct tm ct;
+      #ifdef _WIN32
+      gmtime_s(&ct, &t);
+      #else
+      gmtime_r(&t, &ct);
+      #endif
+      char dateString[128];
+      strftime(dateString, sizeof(dateString), "%a, %d %b %Y %H:%M:%S GMT", &ct);
```

Comment:
Please add an issue to put the code of generating a RFC1123 format into a helper function. Or if it is going to be supported by Azure Core, please add a TODO here.

Reply:
TODO added